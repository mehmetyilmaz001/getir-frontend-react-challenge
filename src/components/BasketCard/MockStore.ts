import { Basket, BasketItem } from './../../types/Basket';
export const mockBasketItem: BasketItem = {
    added: 213213123,
    description: "",
    itemType: "",
    manufacturer: "",
    slug: "",
    tags: [],
    name: "test",
    price: 111,
    quantity: 1,
  };

export const mockBaskeCardStore: Basket = {
    items: [mockBasketItem],
    total: 2222
}