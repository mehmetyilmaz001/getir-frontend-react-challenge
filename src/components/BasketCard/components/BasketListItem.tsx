import { FunctionComponent } from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { addOrIncreaseItem, removeOrDecreaseItem } from '../../../redux/reducers/BasketReducer';
import theme from '../../../style/Theme';
import { BasketItem} from '../../../types/Basket';
import Button from '../../common/Button/Button';
import FormattedLabel from '../../common/FormattedLabel/FormattedLabel';

const BasketItemContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    position: relative;

    :after {
        content: "";
        position: absolute;
        bottom: -10px;
        width: 100%;
        height: 1px;
        background-color: ${theme.palette.border.main};
    }
`;

const TitleAndPrice = styled.div`
    display: flex;
    flex: 3;
    flex-direction: column;
    gap: ${theme.spacing(1)};

    h5 {margin: 0;}
`;

const Buttons = styled.div`
    display: flex;
    flex-direction: row;
    flex: 1;
    justify-content: space-between;
    align-items: center;
    
`;

interface BasketItemProps {
    item: BasketItem
}
 
const BasketListItem: FunctionComponent<BasketItemProps> = ({item}) => {
    const dispatch = useDispatch();
    return (
        <BasketItemContainer>
            <TitleAndPrice>
                <h5>{item.name}</h5>
                <FormattedLabel price={item.price} />
            </TitleAndPrice>

            <Buttons>
                <Button customType='transparent' onClick={() => dispatch(removeOrDecreaseItem(item))}>-</Button>
                <div>{item.quantity}</div>
                <Button customType='transparent' onClick={() => dispatch(addOrIncreaseItem(item))}>+</Button>
            </Buttons>
        </BasketItemContainer>
      );
}

export default BasketListItem;

