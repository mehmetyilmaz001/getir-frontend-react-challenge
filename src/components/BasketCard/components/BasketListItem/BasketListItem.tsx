import { FunctionComponent } from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { addOrRemoveItem } from '../../../../redux/reducers/BasketReducer';
import theme, { colors } from '../../../../style/Theme';
import { BasketItem} from '../../../../types/Basket';
import Button from '../../../common/Button/Button';
import FormattedLabel from '../../../common/FormattedLabel/FormattedLabel';
import { ReactComponent as MinusIcon } from '../../../common/assets/minus.svg';
import { ReactComponent as PlusIcon } from '../../../common/assets/plus.svg';

const BasketItemContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    position: relative;
    margin-bottom: 20px;
    width: 100%;

    :after {
        content: "";
        position: absolute;
        bottom: -18.39px;
        width: 100%;
        height: 1px;
        background-color: ${theme.palette.border.main};
    }
`;

const TitleAndPrice = styled.div`
    display: flex;
    flex: 3;
    flex-direction: column;
    gap: ${theme.spacing(1)};

    h5 {margin: 0; font-size: 14px}
`;


const Price = styled(FormattedLabel)`
    font-size:14px;
    font-weight: 600;
    color: ${theme.palette.primary.main};
`;

const Buttons = styled.div`
    display: flex;
    flex-direction: row;
    flex: 1;
    justify-content: space-between;
    align-items: center;
`;

const Quaintity = styled.div`
    display: flex;
    background-color: ${theme.palette.primary.main};
    color: ${colors.white};
    justify-content: center;
    align-items: center;
    width: 32px;
    height: 32px;
    font-weight: 700;
`;

interface BasketItemProps {
    item: BasketItem
}
 
const BasketListItem: FunctionComponent<BasketItemProps> = ({item}) => {
    const dispatch = useDispatch();
    return (
        <BasketItemContainer>
            <TitleAndPrice>
                <h5>{item.name}</h5>
                <Price price={item.price} />
            </TitleAndPrice>

            <Buttons>
                <Button customType='transparent' onClick={() => dispatch(addOrRemoveItem(item, 'remove'))}><MinusIcon /></Button>
                <Quaintity>{item.quantity}</Quaintity>
                <Button customType='transparent' onClick={() => dispatch(addOrRemoveItem(item, 'add'))}><PlusIcon /></Button>
            </Buttons>
        </BasketItemContainer>
      );
}

export default BasketListItem;

