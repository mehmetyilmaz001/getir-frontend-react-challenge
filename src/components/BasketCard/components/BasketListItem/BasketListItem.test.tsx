import BasketListItem from "./BasketListItem";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { mockBasketItem } from "../../MockStore";

const mockStore = configureStore([]);

test("Basket list item renders correctly", async () => {
  let store;
  let component;

  store = mockStore({
    total: 111,
  });


  component = renderer.create(
    <Provider store={store}>
      <BasketListItem item={mockBasketItem} />
    </Provider>
  );
  expect(component!.toJSON()).toMatchSnapshot();
});
