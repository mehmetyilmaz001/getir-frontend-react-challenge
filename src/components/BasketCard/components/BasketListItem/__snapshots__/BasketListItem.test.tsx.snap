// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket list item renders correctly 1`] = `
<div
  className="sc-gsDKAQ iZHfAx"
>
  <div
    className="sc-dkPtRN dOwCeb"
  >
    <h5>
      test
    </h5>
    <span
      className="sc-hKwDye jFULnG"
    >
      â‚º111,00
    </span>
  </div>
  <div
    className="sc-eCImPb bycmXI"
  >
    <button
      className="sc-bdvvtL bLUCpk"
      onClick={[Function]}
    >
      <div />
    </button>
    <div
      className="sc-jRQBWg gIqxUt"
    >
      1
    </div>
    <button
      className="sc-bdvvtL bLUCpk"
      onClick={[Function]}
    >
      <div />
    </button>
  </div>
</div>
`;
